test/language/statements/async-generator/yield-star-async-next.js

```diff
@@ -0,0 +1,196 @@
+/*---
+ author: Tooru Fujisawa [:arai] <arai_a@mac.com>
+ esid: pending
+ description: execution order for yield* with async iterator and next()
+ info: >
+    YieldExpression: yield * AssignmentExpression
+
+    ...
+    2. Let value be ? GetValue(exprRef).
+    3. Let generatorKind be ! GetGeneratorKind().
+    4. Let iterator be ? GetIterator(value, generatorKind).
+    5. Let received be NormalCompletion(undefined).
+    6. Repeat
+      a. If received.[[Type]] is normal, then
+        i. Let innerResult be ? IteratorNext(iterator, received.[[Value]]).
+        ii. Let innerResult be ? Invoke(iterator, "next",
+            « received.[[Value]] »).
+        iii. If generatorKind is async, then set innerResult to
+             ? Await(innerResult).
+        ...
+        v. Let done be ? IteratorComplete(innerResult).
+        vi. If done is true, then
+           1. Return ? IteratorValue(innerResult).
+        vii. Let received be GeneratorYield(innerResult).
+      ...
+
+    GetIterator ( obj [ , hint ] )
+
+    ...
+    3. If hint is async,
+      a. Set method to ? GetMethod(obj, @@asyncIterator).
+      b. If method is undefined,
+        i. Let syncMethod be ? GetMethod(obj, @@iterator).
+        ii. Let syncIterator be ? Call(syncMethod, obj).
+        iii. Return ? CreateAsyncFromSyncIterator(syncIterator).
+    ...
+
+    GeneratorYield ( iterNextObj )
+
+    ...
+    10. If generatorKind is async,
+      a. Let value be IteratorValue(iterNextObj).
+      b. Let done be IteratorComplete(iterNextObj).
+      c. Return ! AsyncGeneratorResolve(generator, value, done).
+    ...
+
+ flags: [async]
+---*/
+
+var log = [];
+var iter = {
+  get [Symbol.iterator]() {
+    log.push({ name: "get [Symbol.iterator]" });
+  },
+  get [Symbol.asyncIterator]() {
+    log.push({ name: "get [Symbol.asyncIterator]" });
+    return function(...args) {
+      log.push({
+        name: "call [Symbol.asyncIterator]",
+        args: args
+      });
+      var nextCount = 0;
+      return {
+        get next() {
+          log.push({ name: "get next" });
+          return function(...args) {
+            log.push({
+              name: "call next",
+              args: args
+            });
```

Comment:
Do you think it's better checking thisValue for `next`, `return`, `throw`, and `then` ?
currently those functions' thisValue (object) are not global variable and cannot compare directly.
and putting them on global will make the test hard to follow, since the state (`nextCount` etc) needs to also be global.

would it be fine to define `name` property or something on those objects and compare `thisValue.name` against them?